src/routers/datatypeRouters/edges/variants_coding_variants.ts

```diff
@@ -0,0 +1,116 @@
+import { z } from 'zod'
+import { db } from '../../../database'
+import { publicProcedure } from '../../../trpc'
+import { loadSchemaConfig } from '../../genericRouters/genericRouters'
+import { getDBReturnStatements, getFilterStatements, paramsFormatType } from '../_helpers'
+import { QUERY_LIMIT } from '../../../constants'
+import { descriptions } from '../descriptions'
+import { TRPCError } from '@trpc/server'
+import { findVariants, singleVariantQueryFormat, variantSimplifiedFormat } from '../nodes/variants'
+import { codingVariantsFormat } from '../nodes/coding_variants'
+
+const MAX_PAGE_SIZE = 500
+
+const schema = loadSchemaConfig()
+
+const variantCodingVariantSchema = schema['variants to coding variant']
+const variantSchema = schema['sequence variant']
+const codingVariantSchema = schema['coding variant']
+
+const codingVariantsQueryFormat = z.object({
+  name: z.string().optional(),
+  hgvsp: z.string().optional(),
+  page: z.number().default(0),
+  limit: z.number().optional()
+})
+
+function validateInput (input: paramsFormatType): void {
+  if (input.spdi === undefined && input.hgvs === undefined && input.variant_id === undefined) {
+    throw new TRPCError({
+      code: 'BAD_REQUEST',
+      message: 'At least one parameter must be defined.'
+    })
+  }
+}
+
+async function findCodingVariants (input: paramsFormatType): Promise<any[]> {
+  validateInput(input)
+
+  input.page = 0
+  const variant = (await findVariants(input))
+  if (variant.length === 0) {
+    throw new TRPCError({
+      code: 'NOT_FOUND',
+      message: 'Variant not found.'
+    })
+  }
+
+  let limit = QUERY_LIMIT
+  if (input.limit !== undefined) {
+    limit = (input.limit as number <= MAX_PAGE_SIZE) ? input.limit as number : MAX_PAGE_SIZE
+    delete input.limit
+  }
+
+  const query = `
+    FOR record IN ${variantCodingVariantSchema.db_collection_name}
+    FILTER record._from == 'variants/${variant[0]._id}'
+    SORT record._key
+    LIMIT ${input.page as number * limit}, ${limit}
+    RETURN
+      (
+        FOR otherRecord in ${codingVariantSchema.db_collection_name}
+        FILTER otherRecord._id == record._to
+        RETURN {${getDBReturnStatements(codingVariantSchema).replaceAll('record', 'otherRecord')}}
+      )[0]
+  `
+
+  return await (await db.query(query)).all()
+}
+
+async function findVariantsFromCodingVariants (input: paramsFormatType): Promise<any[]> {
+  let limit = QUERY_LIMIT
```

Comment:
Do you need to validate the input here as well? If yes then please validate the input first. If no then all good!